pymc/util.py

```diff
@@ -239,18 +239,18 @@ def enhanced(*args, **kwargs):
 
 
 def dataset_to_point_list(
-    ds: xarray.Dataset, sample_dims: Sequence[str]
+    ds: Union[xarray.Dataset, dict[str, xarray.DataArray]], sample_dims: Sequence[str]
 ) -> Tuple[List[Dict[str, np.ndarray]], Dict[str, Any]]:
     # All keys of the dataset must be a str
     var_names = list(ds.keys())
```

Comment:
```suggestion
    var_names = cast(List[str], list(ds.keys()))
```

might that be ok and make mypy happy? It looks like it know enough to know ds.keys() contains hashables, but not so much to see that the next line checks they are specifically strings

Reply:
That's definitely okay by me